server/src/main/java/org/elasticsearch/index/fielddata/FieldDataContext.java

```diff
@@ -30,7 +40,9 @@ public record FieldDataContext(String fullyQualifiedIndexName, Supplier<SearchLo
     public static FieldDataContext noRuntimeFields(String reason) {
         return new FieldDataContext(
             "",
-            () -> { throw new UnsupportedOperationException("Runtime fields not supported for [" + reason + "]"); }
+            () -> { throw new UnsupportedOperationException("Runtime fields not supported for [" + reason + "]"); },
+            null,
```

Comment:
It might be safer to have a default sourcePathsLookup of `s -> Set.of(s)` here?

Reply:
Yes! Fixed.